File: Ejemplos/Octave/secante.m
function secante()
  
  clc;clear;
  
  f = 'exp(-x**2)-x';
  x0 = 0;
  x1 = 1;
  tol = 0.00001;
  iterMax = 1000;
  
  [xk k error] = Secante(f,x0,x1,tol,iterMax)
  
  
endfunction

function [xk k error] = Secante(f,x0,x1,tol,iterMax)
  
  pkg load symbolic
  
  f1 = matlabFunction(sym(f));

  
  for k=1:iterMax
    
    n = f1(x1) * (x1-x0);
    d = f(x1)-f(x0);
    
    xk = x1 -(n/d);
    
    x0 = x1;
    x1 = xk;
    
    error = abs(f1(xk));
    
    if error<tol
          break
        endif
        
  endfor
  
endfunction

File: Ejemplos/Python/secante.py
import numpy as np
import sympy as sp

def secante(f,x0,x1,tol,iterMax):

    x = sp.Symbol('x')
    f1 = sp.sympify(f)

    k = 0
    error = tol + 1

    while k < iterMax:

        fx0 = sp.N(f1.subs(x,x0))
        fx1 = sp.N(f1.subs(x,x1))

        xk = x1 - fx1*((x1-x0) / (fx1 - fx0))

        x0 = x1
        x1 = xk

        error = abs(f1.subs(x,xk))

        if error<tol:
            break

        k = k+1

    return [xk , k , error]


f = 'exp(-x**2)-x'
x0 = 0
x1 = 1
tol = 10**-5
iterMax = 1000

y = secante(f,x0,x1,tol,iterMax)
print(y)
